app/controllers/groups_controller.rb

```diff
@@ -294,6 +327,23 @@ def add_members
     end
   end
 
+  def handle_membership_request
+    group = Group.find_by(id: params[:id])
+    guardian.ensure_can_edit!(group)
+
+    user = User.find_by(id: params[:user_id])
+    raise Discourse::InvalidParameters.new(:user_id) if user.blank?
+
+    if params[:accept]
+      group.add(user)
+      GroupActionLogger.new(current_user, group).log_add_user_to_group(user)
+    end
+
+    GroupRequest.where(group_id: group.id, user_id: user.id).delete_all
```

Comment:
I think the addition of the user of the group + logging + deletion of the group request should happen in the same DB transaction.